core/primitives/src/block.rs

```diff
@@ -245,7 +252,13 @@ impl Block {
                 + 2 * *gas_price_adjustment_rate.numer() as u128 * u128::from(gas_used);
             let denominator =
                 2 * *gas_price_adjustment_rate.denom() as u128 * u128::from(gas_limit);
-            prev_gas_price * numerator / denominator
+            let new_gas_price =
+                U256::from(prev_gas_price) * U256::from(numerator) / U256::from(denominator);
```

Comment:
BTW in https://github.com/nearprotocol/nearcore/pull/2813 I'm using `BigUInt` from `num_bigint`, which is part of `num` crate family. I think it's better than `U256` even though `U256` is enough here.

Reply:
why do you think it's better than `U256`? Is it faster?

Reply:
@bowenwang1996 Because it's not limited to a fixed size. But `U256` is faster here, because you know it's not going to overflow in this case.